### PR TITLE
Replace default mayavi modules by custom modules

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

File: simphony_ui/ui.py [hunk 1]
```diff
@@ -203,20 +202,65 @@ def show_liggghts_result(self):
...
+            [numpy.linalg.norm(particle.data[CUBA.VELOCITY])
+             for particle
+             in dataset.iter_particles()]
+        )
+        max_velocity = numpy.max(velocities)
+
+        # If the max velocity is not 0, we create the arrow glyph
+        if max_velocity != 0:
+            # Compute max size of particles
+            max_radius = numpy.max(numpy.array(
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: If there's no radius, it will throw an exception.

File: simphony_ui/ui.py [hunk 2]
```diff
@@ -203,20 +202,65 @@ def show_liggghts_result(self):
...
+            1
+
+        # Create Arrow glyph
+        # Get maximum particle velocity
+        velocities = numpy.array(
+            [numpy.linalg.norm(particle.data[CUBA.VELOCITY])
+             for particle
+             in dataset.iter_particles()]
+        )
+        max_velocity = numpy.max(velocities)
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: What happens if max_velocity is zero?

File: simphony_ui/ui.py [hunk 3]
```diff
@@ -203,20 +202,65 @@ def show_liggghts_result(self):
...
+        if max_velocity != 0:
+            # Compute max size of particles
+            max_radius = numpy.max(numpy.array(
+                [particle.data[CUBA.RADIUS]
+                 for particle
+                 in dataset.iter_particles()]
+            ))
+
+            # Compute scale factor of arrows
+            arrow_scale_factor = 20.0 * max_radius/max_velocity
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: if max_radius is zero, you will obtain a scale factor of zero.

**Review by stefanoborini — CHANGES_REQUESTED** [review]

File: simphony_ui/ui.py [hunk 4]
```diff
@@ -203,20 +202,65 @@ def show_liggghts_result(self):
...
+
+        # Add sphere glyph module
+        mayavi_engine.add_module(sphere_glyph_module)
+
+        sphere_glyph_module.glyph.glyph_source.glyph_source = \
+            SphereSource()
+        sphere_glyph_module.glyph.scale_mode = 'scale_by_scalar'
+        sphere_glyph_module.glyph.glyph.range = [0, 1]
+        sphere_glyph_module.glyph.glyph_source.glyph_source.radius = \
+            1
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: 1.0

**Review by stefanoborini — CHANGES_REQUESTED** [review]

File: simphony_ui/ui.py [hunk 5]
```diff
@@ -203,20 +202,65 @@ def show_liggghts_result(self):
...
+
+        source.point_vectors_name = 'VELOCITY'
+
+        # Add sphere glyph module
+        mayavi_engine.add_module(sphere_glyph_module)
+
+        sphere_glyph_module.glyph.glyph_source.glyph_source = \
+            SphereSource()
+        sphere_glyph_module.glyph.scale_mode = 'scale_by_scalar'
+        sphere_glyph_module.glyph.glyph.range = [0, 1]
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: floating points

**Review by stefanoborini — CHANGES_REQUESTED** [review]

File: simphony_ui/ui.py [hunk 6]
```diff
@@ -153,6 +157,9 @@ def show_openfoam_result(self):
         """ Function which add the Openfoam dataset to the
         mayavi scene
         """
+        if self.openfoam_wrapper is None:
+            return
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: If the openfoam_wrapper is None, you probably have to clear the scene as well.

File: simphony_ui/ui.py [hunk 7]
```diff
@@ -166,20 +173,20 @@ def show_openfoam_result(self):
...
-            mayavi_engine.add_module(module)
+        # Add surface module to Openfoam source
+        mayavi_engine.add_module(Surface())
 
     @on_trait_change('liggghts_wrapper')
     def show_liggghts_result(self):
         """ Function which add the Liggghts datasets to the
         mayavi scene
         """
+        if self.liggghts_wrapper is None:
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: same

File: simphony_ui/ui.py [hunk 8]
```diff
@@ -203,20 +208,58 @@ def show_liggghts_result(self):
...
+        )
+        max_velocity = numpy.max(velocities)
 
-        # Add default Liggghts modules
-        for module in flow_modules:
-            mayavi_engine.add_module(module)
+        # If the max velocity is not 0, we create the arrow glyph
+        if max_velocity != 0:
+            # Set scale factor of arrows
+            arrow_scale_factor = 50000.0
```
Comment (stefanoborini) [inline simphony_ui/ui.py]: I think we should write as a comment something like: velocities are in whatever unit. This conversion factor makes 1 graphical unit = x velocity units.

The question is: what is the velocity unit? I'll see if I find it (doubt it, without checking the output)

Comment (stefanoborini) [reply simphony_ui/ui.py]: Openfoam units are metric, so it's meter/second.

**Review by stefanoborini — CHANGES_REQUESTED** [review]

File: simphony_ui/ui.py (hunk 8 above)
Comment (stefanoborini) [reply simphony_ui/ui.py]: which means that we are working in the range of tens of micrometers/sec. I would be tempted to say 1 gu = 1 micrometer/sec. That will make arrows probably good enough for all purposes.

Comment (martinRenou) [reply simphony_ui/ui.py]: Ok thanks :)

**Review by stefanoborini — APPROVED** [review]

**Pull request merged**